models/yang/sonic/sonic-bgp-common.yang

```diff
@@ -0,0 +1,362 @@
+module sonic-bgp-common {
+  namespace "http://github.com/Azure/sonic-bgp-common";
+  prefix sbc;
+
+  import ietf-inet-types {
+    prefix inet;
+  }
+
+  organization
+    "SONiC";
+
+  contact
+    "SONiC";
+
+  description
+    "SONIC BGP common YANG attributes for Neighbors and Peer group";
+
+  revision 2019-11-04 {
+    description
+      "Initial revision.";
+  }
+
+  typedef bgp_peer_type {
+    type enumeration {
+      enum internal {
+        description "Internal peer";
+      }
+      enum external {
+        description "External peer";
+      }
+    }
+  }
+
+  typedef bgp_community_type {
+    type enumeration {
+      enum standard {
+        description "Standard communities";
+      }
+      enum extended {
+        description "Extended communities";
+      }
+      enum both {
+        description "Both standard and extended communities";
+      }
+      enum none {
+        description "No community attribute";
+      }
+    }
+    description
+      "type describing variations of community attributes";
+  }
+
+  typedef sonic_bgp_direction {
+    type enumeration {
+      enum inbound {
+        description
+          "Refers to information received from the BGP peer";
+      }
+      enum outbound {
+        description
+          "Refers to information advertised to the BGP peer";
+      }
+    }
+    description
+        "Type to describe the direction";
+  }
+
+
+  grouping sonic-bgp-cmn {
+
+    leaf vrf_name {
+      type string;
+        description "Network-instance/VRF name";
```

Comment:
Should be leafref to vrf table.

Reply:
yes, we're in the process of doing it, will commit validation checks as part of next PR.